crates/harness/src/test_runner.rs

```diff
@@ -43,15 +43,20 @@ impl TestRunner {
     }
 
     /// Kill the balancer and start a new one with the same configuration.
-    pub async fn restart_balancer(&mut self) {
+    pub async fn restart_balancer(&mut self) -> bool {
```

Comment:
This just makes this function more complex. Is there any reason why you can't do this in a new function?